lib/galaxy/managers/users.py

```diff
@@ -295,6 +295,13 @@ def by_api_key(self, api_key: str, sa_session=None):
             raise exceptions.AuthenticationFailed("Provided API key has expired.")
         return provided_key.user
 
+    def by_oidc_access_token(self, access_token: str):
+        if hasattr(self.app, "authnz_manager") and self.app.authnz_manager:
```

Comment:
It might be more explicit to find which type of app we need (is it `UniverseApplication`) with an isinstance check ?

Reply:
Yeah, it's not great, I've simply followed the pattern used elsewhere. This is because authnz_manager is assigned conditionally even in the UniverseApplication constructor and is undefined by default. This is something that should probably be handled during the proposed refactoring run.